File: scripts/patch_pages_menu.py
# -*- coding: utf-8 -*-
import os
import re

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

def patch(path):
    full = os.path.join(ROOT, path.replace("/", os.sep))
    if not os.path.isfile(full):
        return
    with open(full, "r", encoding="utf-8") as f:
        html = f.read()
    orig = html
    html = re.sub(r'\s*<div data-city-include="tools"></div>\s*', "\n", html)
    if "city-menu.js" not in html and "site-include.js" in html:
        html = html.replace(
            '<script src="',
            '<script src="',
            1,
        )
        html = re.sub(
            r'(<script src="[^"]*site-include\.js"[^>]*></script>)',
            r'\1\n<script src="' + menu_src(path) + 'city-menu.js" defer></script>',
            html,
            count=1,
        )
    if html != orig:
        with open(full, "w", encoding="utf-8", newline="\n") as f:
            f.write(html)
        print("patched", path)

def menu_src(rel_path):
    rel_dir = os.path.dirname(rel_path).replace("\\", "/")
    if not rel_dir:
        return "./"
    return "../" * len([p for p in rel_dir.split("/") if p]) + "js/"

def main():
    for dirpath, _, files in os.walk(ROOT):
        if "scripts" in dirpath or "partials" in dirpath:
            continue
        for f in files:
            if f.endswith(".html"):
                rel = os.path.relpath(os.path.join(dirpath, f), ROOT).replace("\\", "/")
                patch(rel)

if __name__ == "__main__":
    main()

File: scripts/patch-pages-menu.mjs
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const ROOT = path.dirname(path.dirname(fileURLToPath(import.meta.url)));
let patched = 0;

function menuSrc(relPath) {
  const relDir = path.dirname(relPath).replace(/\\/g, "/");
  if (!relDir || relDir === ".") return "./js/";
  const depth = relDir.split("/").filter(Boolean).length;
  return "../".repeat(depth) + "js/";
}

function patchFile(rel) {
  const full = path.join(ROOT, rel);
  if (!fs.existsSync(full)) return;
  let html = fs.readFileSync(full, "utf8");
  const orig = html;
  html = html.replace(/\s*<div data-city-include="tools"><\/div>\s*/g, "\n");
  if (!html.includes("city-menu.js") && html.includes("site-include.js")) {
    const src = menuSrc(rel);
    html = html.replace(
      /(<script src="[^"]*site-include\.js"[^>]*><\/script>)/,
      `$1\n<script src="${src}city-menu.js" defer></script>`
    );
  }
  if (html !== orig) {
    fs.writeFileSync(full, html.replace(/\r\n/g, "\n"), "utf8");
    patched++;
    console.log("patched", rel);
  }
}

function walk(dir) {
  for (const name of fs.readdirSync(dir)) {
    const full = path.join(dir, name);
    const rel = path.relative(ROOT, full).replace(/\\/g, "/");
    if (rel.startsWith("scripts/") || rel.startsWith("partials/")) continue;
    const st = fs.statSync(full);
    if (st.isDirectory()) walk(full);
    else if (name.endsWith(".html")) patchFile(rel);
  }
}

walk(ROOT);
console.log("done", patched);
